backend/tests/api/event/index.js

```diff
@@ -0,0 +1,221 @@
+const request = require('supertest');
+const { jwt } = require("../../helpers/strapi");
+const userFactory = require("../../user/factory");
+
+describe("Event Test", () => {
+    let authenticatedUser;
+    let adminUser;
+    beforeAll(async () => {
+        authenticatedUser = await userFactory.createUser(strapi, "authenticated");
+        adminUser = await userFactory.createUser(strapi, "admin");
+    });
+
+    /**
+     * @param {string | number} id
+     */
+    const constructEvent = (id) => {
+        return {
+            data: {
+                "Title": "Event Title " + id,
+                "Poster": null,
+                "Picture": null,
+                "Description": "Event Description " + id,
+                "Location": "Event Location " + id,
+                "Fee": "Event Fee " + id, 
+                "Contact": "Event Contact " + id,
+                "Private": false,
+                "locale": "en",
+                "publishedAt": Date.now(),
+            }
+        };
+    }
+
+    // Create
+    it("[Create] Public user should not post Events", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/events")
+            .set("accept", "application/json")
+            .send(constructEvent(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("[Create] Authenticated user should not post Events", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/events")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(authenticatedUser.id)}`)
+            .send(constructEvent(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("[Create] Admin user should post Events", async () => {
+        const event = constructEvent(0);
+        await request(strapi.server.httpServer)
+            .post("/api/events")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(adminUser.id)}`)
+            .send(constructEvent(0))
+            .expect("Content-Type", /json/)
+            .expect(200)
+            .expect((data) => {
+                data = data.body.data;
+                expect(data.attributes.Title).toBe(event.data.Title);
+                // expected "null", but received "undefined"
```

Comment:
What do you plan to do with these?

Reply:
I checked Poster and Picture at first but I got that error. Then when I checked Swagger, they did not get Poster and Picture as well. That's why I just commented it out..

@doolri1276 do you know the reason why we don't check Poster and Picture?

Reply:
@jiahn326 Haven't figured out how to test the files. We are skipping the file testing for now :)